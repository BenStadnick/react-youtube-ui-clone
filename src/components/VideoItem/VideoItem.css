.VideoItemCard {
  cursor: pointer;
  width: 95%;
  margin: 10px;
}

.VideoItemCard:hover {
  border-color: green;
}

.ImageContainer {
  text-align: center;
}