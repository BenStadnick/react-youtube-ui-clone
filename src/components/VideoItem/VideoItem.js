import React, {useState, useRef, useEffect}  from 'react';
import Card from 'react-bootstrap/Card'
import './VideoItem.css'

const VideoItem = ({ video, onVideoSelect }) => {
  const ref = useRef(null);

  const [elemWidth, setElemWidth] = useState(0);

  const resize = () => { // need to throttle
    setElemWidth(ref.current ? ref.current.offsetWidth : 0);
  }

  useEffect(() => {
    window.addEventListener('resize', resize);
    setElemWidth(ref.current ? ref.current.offsetWidth : 0);
  }, []);

  if(!video) return <div>Unalbe to load video</div>

  return (
    <Card className='VideoItemCard' onClick={() => onVideoSelect(video)}>
      <Card.Body>
        <div ref={ref} className='ImageContainer'>
          <img style={{ maxHeight: `${elemWidth*0.5625}px`, maxWidth: `${elemWidth}px` }} src={video.snippet.thumbnails.medium.url} alt="Unable to load thumbnail" />
        </div>
        <Card.Text>{video.snippet.title}</Card.Text>
      </Card.Body>
    </Card>
  )
}

export default VideoItem;