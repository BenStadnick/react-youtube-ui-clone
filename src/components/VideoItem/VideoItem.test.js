import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, mount} from 'enzyme';
import VideoItem from './VideoItem';

const mockOnVideoSelect = jest.fn();

 // mock video api response
 const testTitle = 'test title';
 const testUrl = 'test url';
 const video = {
   snippet : {
     title : testTitle,
     thumbnails : {
       medium : {
         url : testUrl
       }
     }
   }
 };

describe('VideoItem component', () => {
  it('did shallow render', () => {
    let wrapper = shallow(<VideoItem />);
    expect(wrapper).toMatchSnapshot();

    wrapper = shallow(<VideoItem video={video} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays correct thumbnail and title', () => {
    const wrapper = mount(<VideoItem video={video} />);

    const img = wrapper.find('img');
    expect(img.props().src).toEqual(testUrl);

    const title = wrapper.find('CardText');
    expect(title.text()).toEqual(testTitle);

    wrapper.unmount();
  });

  it('calls onVideoSelect when clicked', () => {
    const wrapper = mount(<VideoItem onVideoSelect={mockOnVideoSelect} video={video}/>);

    wrapper.find('Card').simulate('click');
    expect(mockOnVideoSelect).toHaveBeenCalled();
    
    wrapper.unmount();
  });

});