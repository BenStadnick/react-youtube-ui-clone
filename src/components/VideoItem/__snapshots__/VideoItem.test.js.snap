// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItem component did shallow render 1`] = `
<div>
  Unalbe to load video
</div>
`;

exports[`VideoItem component did shallow render 2`] = `
<Card
  body={false}
  className="VideoItemCard"
  onClick={[Function]}
>
  <CardBody>
    <div
      className="ImageContainer"
    >
      <img
        alt="Unable to load thumbnail"
        src="test url"
        style={
          Object {
            "maxHeight": "0px",
            "maxWidth": "0px",
          }
        }
      />
    </div>
    <CardText>
      test title
    </CardText>
  </CardBody>
</Card>
`;
